swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/SubscriptionDefinition.java

```diff
@@ -173,4 +180,47 @@ public static Optional<SubscriptionDefinition> lookupSubscriptionByRole(String r
                     .anyMatch(variant -> variant.getRoles().contains(role)))
         .findFirst();
   }
+
+  /**
+   * Looks for tag matching a variant
+   *
+   * @param tag
+   * @return Optional<Subscription>
+   */
+  public static Optional<SubscriptionDefinition> lookupSubscriptionByTag(
+      @NotNull @NotEmpty String tag) {
+
+    return SubscriptionDefinitionRegistry.getInstance().getSubscriptions().stream()
+        .filter(subscription -> !subscription.getVariants().isEmpty())
+        .filter(
+            subscription ->
+                subscription.getVariants().stream()
+                    .anyMatch(variant -> Objects.equals(tag, variant.getTag())))
+        .collect(MoreCollectors.toOptional());
```

Comment:
can we avoid changing the pattern by doing `.findFirst()` instead of `.collect(MoreCollectors.toOptional());`

Reply:
good point will fix that and remove dependency. thanks!